packages/rum-core/src/domain/view/viewMetrics/trackScrollMetrics.spec.ts

```diff
@@ -1,23 +1,36 @@
-import type { RelativeTime, TimeStamp, Duration } from '@datadog/browser-core'
+import type { Duration, RelativeTime, TimeStamp } from '@datadog/browser-core'
 import { DOM_EVENT } from '@datadog/browser-core'
-import type { Clock } from '@datadog/browser-core/test'
-import { createNewEvent, mockClock } from '@datadog/browser-core/test'
+import { collectAsyncCalls, createNewEvent } from '@datadog/browser-core/test'
 import type { TestSetupBuilder } from '../../../../test'
 import { setup } from '../../../../test'
-import { PAGE_ACTIVITY_END_DELAY, PAGE_ACTIVITY_VALIDATION_DELAY } from '../../waitPageActivityEnd'
 import type { RumConfiguration } from '../../configuration'
-import { THROTTLE_VIEW_UPDATE_PERIOD } from '../trackViews'
 import type { ViewTest } from '../setupViewTest.specHelper'
 import { setupViewTest } from '../setupViewTest.specHelper'
-import { THROTTLE_SCROLL_DURATION, type ScrollMetrics, trackScrollMetrics } from './trackScrollMetrics'
-
-const BEFORE_PAGE_ACTIVITY_VALIDATION_DELAY = (PAGE_ACTIVITY_VALIDATION_DELAY * 0.8) as Duration
-
-const AFTER_PAGE_ACTIVITY_END_DELAY = PAGE_ACTIVITY_END_DELAY * 1.1
+import { trackScrollMetrics } from './trackScrollMetrics'
 
 describe('trackScrollMetrics', () => {
   let setupBuilder: TestSetupBuilder
   let viewTest: ViewTest
+  let stopTrackScrollMetrics: () => void
+  let callbackSpy: jasmine.Spy
+  let assertCallbackSpyEqual: (expected: any, done: DoneFn, times?: number) => void
+  const getMetrics = jasmine.createSpy('getMetrics')
+
+  const mockGetMetrics = (scrollParams: { scrollHeight: number; scrollDepth: number; scrollTop: number }) => {
+    getMetrics.and.returnValue(scrollParams)
+  }
+
+  const newScroll = (scrollParams: { scrollHeight: number; scrollDepth: number; scrollTop: number }) => {
+    mockGetMetrics(scrollParams)
+    window.dispatchEvent(createNewEvent(DOM_EVENT.SCROLL))
+  }
+
+  const increaseHeight = (scrollParams: { scrollHeight: number; scrollDepth: number; scrollTop: number }) => {
+    mockGetMetrics(scrollParams)
+    const node = document.createElement('div')
+    node.style.height = `${scrollParams.scrollHeight}px`
+    document.body.replaceChildren(node)
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** ‏`replaceChildren` isn't supported in IE11